src/main/java/com/cloudera/sparkts/api/java/DateTimeIndexFactory.java

```diff
@@ -110,6 +110,14 @@ public static IrregularDateTimeIndex irregular(long[] dts, ZoneId zone) {
     }
 
     /**
+     * Creates a HybridDateTimeIndex composed of the given indices using the provided date-time zone.
+     * It is the responsibility of the caller to ensure that all indices have the same ZoneId as zone.
+     */
+    public static HybridDateTimeIndex hybrid(DateTimeIndex[] indices, ZoneId zone) {
```

Comment:
Is the `zone` argument needed at all?  Why not just take the zone used by the first index and throw an exception if any of the other indices use a different one?